src/modules/memory.cpp

```diff
@@ -107,7 +107,12 @@ namespace modules {
 
   bool memory_module::build(builder* builder, const string& tag) const {
     if (tag == TAG_BAR_USED) {
-      builder->node(m_bar_memused->output(m_perc_memused));
+      string m_type = m_conf.get<decltype(m_type)>(name(), "bar-used-type", "RAM");
```

Comment:
This retrieves the variable from the config every single time the module is built, which isn't very efficient. It would be better if m_type was set from the initialization and only checked in `build()`, or create a new tag like `TAG_BAR_USED_SWAP` altogether.

Reply:
done